checkita-api/src/main/scala/org/checkita/api/configGenerator/DdlParser.scala

```diff
@@ -0,0 +1,104 @@
+package org.checkita.api.configGenerator
+
+import scala.collection.mutable
+import scala.collection.mutable.{ListBuffer, Map => MutableMap}
+
+import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
+import org.apache.spark.sql.catalyst.plans.logical.CreateTable
+
+import net.sf.jsqlparser.parser.CCJSqlParserUtil
+import net.sf.jsqlparser.statement.Statement
+import net.sf.jsqlparser.statement.create.table
+
+import org.raiffeisen.checkita.dqf.utils.Logging
+
+object DdlParser extends Logging {
+
+  /**
+   * Parses DDL string to extract metadata about a table and its columns.
+   *
+   * @param ddl The DDL string to be parsed.
+   * @return A tuple containing two maps:
+   *         - The first map contains basic table information, including the table's source name, location, and storage format.
+   *         - The second map categorizes columns by their data types, with each entry listing the columns and their associated attributes.
+   */
+  def parseDDL(ddl: String): (Map[String, String], Map[String, ListBuffer[Map[String, String]]]) = {
+    var sources: MutableMap[String, String] = MutableMap()
+    val columnDict: MutableMap[String, ListBuffer[Map[String, String]]] = MutableMap()
+
+    if (ddl.toLowerCase.contains("stored as")) {
```

Comment:
We have a connection type variable in request. So we can expect hive ddl when connection type is hive, and use other parses otherwise.